src/core/geometry/qgsray3d.h

```diff
@@ -0,0 +1,48 @@
+#ifndef QGSRAY3D_H
+#define QGSRAY3D_H
+
+#include "qgsbox3d.h"
+
+#include <QVector3D>
+
+/**
+ * \ingroup core
+ * A representation of a ray in 3D.
+ *
+ * A ray is composed of an origin point (the start of the ray) and a direction vector.
+ *
+ * \since QGIS 3.18
+ */
+class CORE_EXPORT QgsRay3D
```

Comment:
Can you add some unit tests for this class?

Reply:
Done